otter/grade/containers.py

```diff
@@ -192,6 +192,15 @@ def kill_container():
             scores = pickle.load(f)
 
         scores = scores.to_dict()
+
+        scores_to_process = list(filter(lambda key: type(scores[key]) == dict, scores))
+        possible_points = 0
+        raw_score = 0
+        for q in scores_to_process:
+            possible_points += scores[q]["possible"]
+            raw_score += scores[q]["score"]
+        scores["percent_correct"] = round(raw_score / possible_points, 4)
```

Comment:
So before line 194 `scores` should be an instance of `otter.test_files.GradingResults` which has properties for accessing points earned and points possible. I'd prefer to use those to calculate the percentage so that we keep all that logic in one place. I think you can rename the left side of the assignment on line 194 and then use something like

```python
scores_dict = scores.to_dict() # this is line 194, need to update all uses of scores below here to use scores_dict instead
scores_dict["percent_correct"] = scores.total / scores.possible
```

Reply:
and yes -- did not see that there was state in the scores instance!